### PR TITLE
Create logger abstractions

### PR DESCRIPTION
**Pull request by sugarraysam**
I find many of our tooling need to rewrite basic logger setup code. This is an attempt at standardizing logging across codebases.

Added both a text logger (for CLI) and a json logger (for openshift based apps).

Also bumping python to 3.8 (3.7 does not exist :thinking:  -> https://catalog.redhat.com/software/containers/) because I encounter segfault issues with pipenv and python 3.6.

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.15.0
+0.16.0
```
Comment (apahim) [inline VERSION:1]: this is a breaking change... people using this lib from a project using python < 3.8 will be broken, let's increment the major version then

**Comment by apahim** [comment]
The release commit should be detached from the code changes, as a release can ship one or more features/fixes.
Other than that, code looks good, thanks you for the utilities.

**Review by apahim — CHANGES_REQUESTED** [review]
Let's not make the release commit part of this MR. Other than that, looks good :)

File: VERSION (hunk 1 above)
Comment (sugarraysam) [reply VERSION]: reverted the commit, will create new MR with major version bump

Comment (rporres) [reply VERSION]: thx for making this a major. We are working in moving qontract-reconcile to a newer Python version, still not there 😓

**Review by apahim — APPROVED** [review]
lgtm

**Pull request merged**